### PR TITLE
[FLINK-7410] [table] Use toString method to display operator names for UserDefinedFunction

### PR DESCRIPTION
**Pull request by hequn8128**
## What is the purpose of the change

*Use toString method to display operator names for UserDefinedFunction. The method will return class name by default. Users can also override the method to return whatever he wants*

## Brief change log

  - *Add `toString` method to `UserDefinedFunction ` and returns class name by default*
  - *Override `toString` method in `AggSqlFunction `, `ScalarSqlFunction ` and `TableSqlFunction ` to return class name*
  - *Modify test cases*

## Verifying this change

This change added tests and can be verified as follows:

  - *Modify CalcTest/CorrelateTest/AggTest to validate that operator names are displayed correctly*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala [hunk 1]
```diff
@@ -41,7 +41,7 @@ abstract class UserDefinedFunction extends Serializable {
   def close(): Unit = {}
 
   /**
-    * @return true iff a call to this function is guaranteed to always return
+    * @return true if a call to this function is guaranteed to always return
```
Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala]: I think `iff` is correct. `iff` is means "if and only if".

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala [hunk 2]
```diff
@@ -268,6 +269,7 @@ object UserDefinedFunctionUtils {
     */
   def createTableSqlFunctions(
       name: String,
+      disPlayName: String,
```
Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala]: disPlayName  -> displayName

**Comment by wuchong** [comment]
The CI failed but is not related to this PR.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala (hunk 1 above)
Comment (greghogan) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala]: We should be writing out "if and only if" iff that is the intent.

Comment (hequn8128) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala]: thanks all, I think "if and only if" is better

**Comment by hequn8128** [comment]
@wuchong  Thanks for your review, I have updated the PR according to your comments.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/sql/CorrelateTest.scala [hunk 3]
```diff
@@ -41,7 +41,8 @@ class CorrelateTest extends TableTestBase {
         "DataSetCorrelate",
         batchTableNode(0),
         term("invocation", "func1($cor0.c)"),
-        term("function", func1.getClass.getCanonicalName),
+        term("correlate", s"table(func1($$cor0.c))"),
+        term("select", "a,b,c,f0"),
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/sql/CorrelateTest.scala]: please change to `term("select", "a", "b", "c", "f0"),` (i.e., use separate strings for the field names) for consistency

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/sql/CorrelateTest.scala [hunk 4]
```diff
@@ -61,7 +62,8 @@ class CorrelateTest extends TableTestBase {
         "DataSetCorrelate",
         batchTableNode(0),
         term("invocation", "func1($cor0.c, '$')"),
-        term("function", func1.getClass.getCanonicalName),
+        term("correlate", s"table(func1($$cor0.c,'$$'))"),
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/sql/CorrelateTest.scala]: change to `term("correlate", s"table(func1($$cor0.c, '$$'))"),` for consistency

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/CommonCorrelate.scala [hunk 5]
```diff
@@ -183,17 +183,25 @@ trait CommonCorrelate {
...
-    val udtfName = sqlFunction.getName
-    val operands = rexCall.getOperands.asScala.map(_.toString).mkString(",")
+  private[flink] def correlateToString(
+      inputType: RelDataType,
+      rexCall: RexCall,
+      sqlFunction: TableSqlFunction,
+      expression: (RexNode, List[String], Option[List[RexNode]]) => String): String = {
+    val inFields = inputType.getFieldNames.asScala.toList
+    val udtfName = sqlFunction.toString
+    val operands = rexCall.getOperands.asScala.map(expression(_, inFields, None)).mkString(",")
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/CommonCorrelate.scala]: please add a space: `mkString(",")` -> `mkString(", ")`

Comment (fhueske) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/CommonCorrelate.scala]: Please add a space to the `mkString` call in `selectToString()` as well. Thanks

**Comment by hequn8128** [comment]
Hi @fhueske , 
Thanks for your review, I have addressed all your comments and rebased the code to the master :)

**Comment by fhueske** [comment]
Merging

**Pull request closed without merging**